### DISCUSSION TITLE
How to customize search to bring us closer to and highlight the searched words on the page?

### DISCUSSION THREAD

**Discussion by EmbraceLife** [quarto-dev/quarto-cli#1750] [Q&A] [answered]
Currently, when we use the search bar to search the website built with quarto, after we click on a result, we are sent to the heading level 1 or level2 of the page where the searched word found. Here are two issues I would like to improve: 

1. could I customize in quarto so that the result link send me to the lowest heading level, like heading level 3 or 4 or even lower where the searched word actually appeared, not the highest heading level, like heading level 2? 
2. could I customize in quarto so that after clicked a result link, the searched word is still highlighted in the page, so that I can scroll through the page to quickly find the searched word, especially when I am not sent to the lowest/closest heading where the search word is actually in?

Thanks a lot!

**Comment by mcanouil** [comment]
Did you take look at <https://quarto.org/docs/websites/website-search.html>, especially the section "Using Algolia" _i.e._, this is what <https://quarto.org> is using, but words do not stay highlighted (web browser search tool can help once you are on the right page)?

**Comment by dragonstyle** [comment, marked as answer]
> could I customize in quarto so that the result link send me to the lowest heading level, like heading level 3 or 4 or even lower where the searched word actually appeared, not the highest heading level, like heading level 2?

Right now, this is hard coded to break the document up at the `h2` level (creating a index entry at for each `h2`). There is, of course, a balance between the size the result and the number of results (too high of a heading level could result in a large number of very small results which are likely to be worse for the user). Not sure the best thing to do here, but feel free to open an issue to make this customizable and we'll give it some thought.

> could I customize in quarto so that after clicked a result link, the searched word is still highlighted in the page, so that I can scroll through the page to quickly find the searched word, especially when I am not sent to the lowest/closest heading where the search word is actually in?

This, too, is not currently supported, though as with the above, feel free to open an issue and we'll see what we can do!
